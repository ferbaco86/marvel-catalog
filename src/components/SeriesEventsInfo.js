import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
// import HeroCard from './HeroCard';

const SeriesEventsContainer = styled.div`
margin-top: 2rem;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;`;

const SubTitle = styled.h3`
color: #202020;
padding: 1rem;
text-transform: uppercase;
font-family: 'Roboto Condensed', sans-serif;
line-spacing: 1px;
font-size: 2rem;
text-align: center;`;

// const {
//   name, image, extension,
// } = props;

const SeriesEventsInfo = props => {
  const {
    title,
  } = props;
  return (
    <>
      <SeriesEventsContainer>
        <SubTitle>
          {title}
        </SubTitle>
      </SeriesEventsContainer>
    </>
  );
};

SeriesEventsInfo.propTypes = {
  title: PropTypes.string.isRequired,
  //   // id: PropTypes.number.isRequired,
  // name: PropTypes.string.isRequired,
  // image: PropTypes.string.isRequired,
  // extension: PropTypes.string.isRequired,
};

export default SeriesEventsInfo;
