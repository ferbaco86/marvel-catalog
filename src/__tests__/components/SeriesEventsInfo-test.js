import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import SeriesEventsInfo from '../../components/SeriesEventsInfo';

describe('Rendering component', () => {
  it('creates a SeriesEventsInfo component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <SeriesEventsInfo title="Test Title" />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Test Title');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <SeriesEventsInfo title="Test Title" />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
