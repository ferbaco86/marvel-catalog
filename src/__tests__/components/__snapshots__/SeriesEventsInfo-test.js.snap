// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-bdfBwQ gIGxFT"
>
  <h3
    className="sc-gsTCUz XKkJw"
  >
    Test Title
  </h3>
</div>
`;
